bazel/wasm/wasm.bzl

```diff
@@ -0,0 +1,87 @@
+#  Copyright 2016-2020 Envoy Project Authors
+#  Copyright 2020 Google LLC
+# 
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+# 
+#       http://www.apache.org/licenses/LICENSE-2.0
+# 
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+load("@rules_cc//cc:defs.bzl", "cc_binary")
+
+def _wasm_cc_transition_impl(settings, attr):
+    return {
+        "//command_line_option:cpu": "wasm32",
+        "//command_line_option:crosstool_top": "@proxy_wasm_cpp_sdk//toolchain:emscripten",
+
+        # Overriding copt/cxxopt/linkopt to prevent sanitizers/coverage options leak
+        # into Wasm build configuration
```

Comment:
Is this still a valid comment and action, or was this only relevant inside Envoy repo?

Notably, Emscripten supports sanitizers, so clearing all compiler and linker options seems feels weird.

Reply:
I don't think the same set of sanitizer / linkopt that would passed to native cc would always work here.

it makes more sense to create another toolchain definition and/or another transition if we want to support UBSan / debugger configs like its document says here: https://emscripten.org/docs/debugging/Sanitizers.html

Reply:
Right, I don't think exactly the same set would work (although, Emscripten supports `-fsanitize=address`, `-fsanitize=leak`, `-fsanitize=undefined`, etc.), but this repo is for WebAssembly only, and it doesn't have `.bazelrc` filled with a bunch of options for non-Wasm targets, so there are no `copt/cxxopt/linkopt` for non-Wasm targets that we need to clear here.

Reply:
Again, the goal of this PR is to make this starlark rule available to wider users, so we do need to clear (or better, modify and that can be a future TODO). If that's not the goal we don't even need this and we can have everything in `.bazelrc`.